gammapy/estimators/ts_map.py

```diff
@@ -198,45 +201,44 @@ def estimate_kernel(self, dataset):
             Kernel map
 
         """
-        # TODO: further simplify the code below
         geom = dataset.exposure.geom
 
+        if self.kernel_width is not None:
+            geom = geom.to_odd_npix(max_radius=self.kernel_width / 2)
+
         model = self.model.copy()
         model.spatial_model.position = geom.center_skydir
 
-        geom_kernel = geom.to_odd_npix(max_radius=self.kernel_width / 2)
-
         # Creating exposure map with exposure at map center
-        exposure = Map.from_geom(geom_kernel, unit="cm2 s1")
-
-        exposure.data[...] = dataset.exposure.to_region_nd_map(geom.center_skydir).data
+        exposure = Map.from_geom(geom, unit="cm2 s1")
+        exposure_center = dataset.exposure.to_region_nd_map(geom.center_skydir)
+        exposure.data[...] = exposure_center.data
 
         # We use global evaluation mode to not modify the geometry
-        evaluator = MapEvaluator(model, evaluation_mode="global")
+        evaluator = MapEvaluator(model=model)
 
         evaluator.update(
-            exposure, dataset.psf, dataset.edisp, dataset.counts.geom, dataset.mask_fit,
+            exposure=exposure,
+            psf=dataset.psf,
+            edisp=dataset.edisp,
+            geom=dataset.counts.geom,
+            mask=dataset.mask_image,
         )
-
         kernel = evaluator.compute_npred()
         kernel.data /= kernel.data.sum()
-
-        if (self.kernel_width >= geom.width).any():
```

Comment:
With padding this case cannot appear anymore. Independent of the size of the kernel, the input map is enlarged correspondingly...